src/api/ledger/parse/utils.js

```diff
@@ -4,6 +4,15 @@ const _ = require('lodash');
 const transactionParser = require('ripple-lib-transactionparser');
 const toTimestamp = require('../../../core/utils').toTimestamp;
 const utils = require('../utils');
+const BigNumber = require('bignumber.js');
+
+function adjustQualityForXRP(quality: string, takerGetsCurrency: string,
+    takerPaysCurrency: string) {
+  const shift = (takerGetsCurrency === 'XRP' ? 6 : 0)
```

Comment:
> That's certainly simpler

It's also exactly what the new API is trying to do! The only differences are that:
1. we are forced to use `totalPrice` instead of `price` if we want to avoid rounding errors
2. we're using the name `direction` instead of `type`, and I think `direction` is better because it's more specific
3. we're using the name `quantity` instead of `amount`, and `quantity` is the more standard term

In summary, the data structures are:
RippleAPI: direction/quantity/totalPrice
"Textbook": direction/quantity/price
Bitstamp: type/amount/price
rippled: tfSell/takerGets/takerPays (different basis; takerGets/takerPays don't map directly to quantity/totalPrice)

> Do you have any examples I could look at?

It might be hard to find examples, but you can easily find explanations, like here: "the EUR/USD currency pair is always quoted indirectly. The reason for this is mostly convention." http://www.investopedia.com/ask/answers/06/eurusd.asp